package/MDAnalysis/auxiliary/base.py

```diff
@@ -0,0 +1,187 @@
+            try:
+                self._read_next_step()
+            except StopIteration:
+                break
+        self.ts_rep = self.calc_representative()
+        ts.aux.__dict__[self.names] = self.ts_rep
+        return ts
+        ## currently means that after reading in a timestep, ts_data and
+        ## ts_diffs correspond to that ts but the current step/step_data of 
+        ## the auxreader is the first step 'belonging' of the next ts...
+
+    def first_in_ts(self, ts):
+        """ Check if current step is first step 'belonging' to *ts*
+        Assumes auxiliary *dt* is constant! """
+        if (self.time-(ts.time-ts.dt/2)) < self.dt:
+            return True
+        else:
+            return False
+
+    def step_in_ts(self, ts):
+        """ Check if current step 'belongs' to *ts* """
+        if (self.time-ts.time) <= ts.dt/2. and (self.time-ts.time) > -ts.dt/2.:
+            return True
+        else:
+            return False
+           
+    def go_to_ts(self, ts):
+        """ Move to and read auxilairy steps corresponding to *ts* """
+        # Need to define in each auxiliary reader
+        raise NotImplementedError(
+            "BUG: Override go_to_ts() in auxilairy reader!")
+
+    def reset_ts(self):
+        self.ts_data = np.array([])
+        self.ts_diffs = []
+
+    def add_step_to_ts(self, ts_time):
+        if len(self.ts_data) == 0:
+            self.ts_data = [self.step_data]
+        else:
+            self.ts_data = np.append(self.ts_data, [self.step_data], axis=0)
+        self.ts_diffs.append(abs(self.time - ts_time))
+
+    def calc_representative(self):
+        if self.cutoff:
+            ## starting to get nasty, should probably change...
+            cutoff_data = np.array([self.ts_data[i] 
+                                    for i,d in enumerate(self.ts_diffs)
+                                    if d < self.cutoff])
+            cutoff_diffs = [d for d in self.ts_diffs if d < self.cutoff]
+        else:
+            cutoff_data = self.ts_data
+            cutoff_diffs = self.ts_diffs
+        if len(cutoff_data) == 0:
+            value = [] # TODO - flag as missing
+        elif self.represent_ts_as == 'closest':
+            value = cutoff_data[np.argmin(cutoff_diffs),:]
+        elif self.represent_ts_as == 'average':
+            value = np.mean(cutoff_data, axis=0)
+        # TODO - interpolation?
+        return value
+    
+    def close():
+        # Overwrite when reading from file to close open file
+        pass    
+
+    @property
+    def time(self):
+        """ Time in ps of current auxiliary step
+        As read from the auxiliary data, if present; otherwise calcuated
+        as (step - 1) * dt + initial_time
+        """
+        try:
+            return self._time
+        except AttributeError:
+            return (self.step - 1) * self.dt + self.initial_time
+
+    # TODO - add __enter__ and __exit__ methods when reading from file
```

Comment:
In that class, `__enter__` and `__exit__` can just do `pass`. They will have to be overwritten by `AuxFileReader`, though.